src/ui/public/toggle_panel/toggle_panel.js

```diff
@@ -0,0 +1,27 @@
+import { uiModules } from 'ui/modules';
+import template from './toggle_panel.html';
+import 'ui/toggle_button';
+
+const app = uiModules.get('kibana');
+
+app.directive('togglePanel', function () {
+  return {
+    restrict: 'E',
+    replace: true,
+    transclude: true,
+    template: template,
+    scope: {
+      togglePanelId: '@',
+      text: '@',
+      isCollapsed: '=',
+      onToggleButtonClick: '='
```

Comment:
This handler's name really bothers me. It's conflating three concepts here:

- the toggling of the open/close state
- some specific dom node (button here)
- the user interaction (click here)

In my opinion, the handler names should generally follow an `onVerb` pattern, and shouldn't be bound to specific elements or interactions. Events should convey some state modification or component specific interaction. In this case, `onToggle` seems sufficient.

@cjcenizal @snide I'm interested in your thoughts on this too.